### PR TITLE
Tariffs location

### PR DESCRIPTION
**Pull request by SomeOne35719**
-added new Entity
-reworked add Tariff logic
-local deactivation of Location in tariff

### CODE REVIEW

File: core/src/main/java/greencity/controller/SuperAdminController.java [hunk 1]
```diff
@@ -570,4 +567,25 @@ public ResponseEntity<HttpStatus> setLimitsByPriceOfOrder(@Valid @PathVariable L
...
+
+    /**
+     * Controller for activation or deactivation Locations in Tariff depends
+     * on @RequestParam.
+     *
+     * @author Yurii Fedorko
+     */
+    @ApiOperation(value = "Change Tariff Location status")
+    @ApiResponses(value = {
+        @ApiResponse(code = 202, message = HttpStatuses.OK),
```
Comment (VadymMakitra) [inline core/src/main/java/greencity/controller/SuperAdminController.java:598]: Wrong stats code

File: dao/src/main/java/greencity/entity/order/TariffsInfo.java [hunk 2]
```diff
@@ -67,12 +67,15 @@ public class TariffsInfo {
     @ManyToOne
     private Courier courier;
 
+    @OneToMany(mappedBy = "tariffsInfo", cascade = CascadeType.ALL)
+    private Set<TariffLocation> tariffLocations;
+    /*-
     @ManyToMany
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/entity/order/TariffsInfo.java]: Remove comments pls

File: dao/src/main/java/greencity/entity/user/Location.java [hunk 3]
```diff
@@ -42,6 +44,10 @@ public class Location {
     @ManyToOne
     private Region region;
 
+    @OneToMany(mappedBy = "location", cascade = CascadeType.ALL)
+    private Set<TariffLocation> tariffLocations;
+    /*-
     @ManyToMany(mappedBy = "locations", cascade = CascadeType.ALL)
     private List<TariffsInfo> tariffsInfoList;
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/entity/user/Location.java]: Remove comments pls

File: dao/src/main/resources/db/changelog/logs/Dump-DDL-01.06.2022.xml [hunk 4]
```diff
@@ -0,0 +1,900 @@
...
+                     left join ubs_user uu on o.ubs_user_id = uu.id
+                     left join address a on uu.address_id = a.id
+                     left join users u on o.users_id = u.id
+
+                     left join order_additional oa on o.id = oa.orders_id;
+        </createView>
+    </changeSet>
+
+
+<!--checkConstraint payment settlement_date-->
```
Comment (VadymMakitra) [inline dao/src/main/resources/db/changelog/logs/Dump-DDL-01.06.2022.xml:1112]: Remove comments pls

File: service-api/src/main/java/greencity/dto/tariff/AddNewTariffResponseDto.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.Setter;
+
+import java.util.List;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class AddNewTariffResponseDto {
+
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/dto/tariff/AddNewTariffResponseDto.java]: Remove whitespace

File: service/src/main/java/greencity/mapping/GetTariffsInfoDtoMapper.java [hunk 6]
```diff
@@ -16,9 +16,14 @@
 public class GetTariffsInfoDtoMapper extends AbstractConverter<TariffsInfo, GetTariffsInfoDto> {
     @Override
     protected GetTariffsInfoDto convert(TariffsInfo source) {
-        Region region = source.getLocations().iterator().next().getRegion();
-        RegionDto regionDto =
-            RegionDto.builder().regionId(region.getId()).nameEn(region.getEnName()).nameUk(region.getUkrName()).build();
+        Region region = source.getTariffLocations() != null
+            ? source.getTariffLocations().iterator().next().getLocation().getRegion()
+            : null;
+        RegionDto regionDto = null;
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/mapping/GetTariffsInfoDtoMapper.java:22]: Please refactor it using the here ternary operator

File: service/src/main/java/greencity/mapping/GetTariffsInfoDtoMapper.java [hunk 7]
```diff
@@ -16,9 +16,14 @@
...
+        if (region != null) {
+            regionDto = RegionDto.builder().regionId(region.getId()).nameEn(region.getEnName())
+                .nameUk(region.getUkrName()).build();
+        }
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/mapping/GetTariffsInfoDtoMapper.java]: Please refactor it using the here ternary operator

File: service/src/main/java/greencity/mapping/TariffsForLocationDtoMapper.java [hunk 8]
```diff
@@ -14,26 +14,31 @@
...
-            RegionDto.builder().regionId(region.getId()).nameEn(region.getEnName()).nameUk(region.getUkrName()).build();
+    public TariffsForLocationDto convert(TariffsInfo source) {
+        Region region = source.getTariffLocations() != null
+            ? source.getTariffLocations().iterator().next().getLocation().getRegion()
+            : null;
+        RegionDto regionDto = null;
+        if (region != null) {
+            regionDto = RegionDto.builder().regionId(region.getId()).nameEn(region.getEnName())
+                .nameUk(region.getUkrName()).build();
+        }
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/mapping/TariffsForLocationDtoMapper.java]: Please refactor it using the here ternary operator

File: service/src/test/java/greencity/ModelUtils.java [hunk 9]
```diff
@@ -3432,12 +3434,14 @@ public static TariffsInfo getTariffInfoWithLimitOfBags() {
...
             .courier(ModelUtils.getCourier())
             .courierLimit(CourierLimit.LIMIT_BY_AMOUNT_OF_BAG)
-            .locations(Set.of(Location.builder()
-                .id(1L)
-                .region(ModelUtils.getRegion())
-                .nameUk("Київ")
-                .nameEn("Kyiv")
-                .coordinates(ModelUtils.getCoordinates())
+            .tariffLocations(Set.of(TariffLocation.builder()
+                // .tariffsInfo(ModelUtils.getTariffInfoWithLimitOfBags())
```
Comment (VadymMakitra) [inline service/src/test/java/greencity/ModelUtils.java]: Remove comments

**Review by Max877866 — APPROVED** [review]

**Review by NewArtMask — APPROVED** [review]

**Review by VadymMakitra — APPROVED** [review]

**Pull request merged**